src/librustdoc/html/render.rs

```diff
@@ -1550,6 +1553,21 @@ impl Context {
     }
 }
 
+fn compute_path(path: String) -> String {
```

Comment:
Wouldn't it be better to use `canonicalize()` or a URL library instead of writing this from scratch? This does not handle symbolic links for example.